x-pack/plugins/reporting/common/constants.ts

```diff
@@ -104,6 +108,26 @@ export const API_GENERATE_IMMEDIATE = `${API_BASE_URL_V1}/generate/immediate/csv
 // Management UI route
 export const REPORTING_MANAGEMENT_HOME = '/app/management/insightsAndAlerting/reporting';
 
+export const REDIRECT_APP_PATH_HOME =
+  '/app/management/insightsAndAlerting/reporting/r/{reportSavedObjectId}/{locatorIdx}';
+
+/**
+ * A way to get the client side route for the reporting redirect app. We should remove this once we are able to
+ * use the Reporting locator to generate this value for us.
+ *
+ * TODO: Remove this and use a Reporting locator in future
```

Comment:
Can you elaborate a bit on "Reporting locator"?

I think there are pros and cons to having the Reporting redirect app fetch the locator parameters for the job, vs just sending the browser to the redirect app with the location parameters as a variable in the query string. Cutting down the extra fetch at job execution time would be valuable. I really doubt that the query string will ever be too long for Chromium to handle. Also, Reporting does not have to interpret the parameters - any state migration that is needed will still occur.

WDYT?

Reply:
> Can you elaborate a bit on "Reporting locator"?

Yes! This TODO is outdated, I don't think we need to add a Reporting Locator for the redirect app at the moment - since we are not sharing this as something for other apps/users to navigate to - although that might change at which point having a locator would be best.

> Cutting down the extra fetch at job execution time would be valuable. I really doubt that the query string will ever be too long for Chromium to handle. Also, Reporting does not have to interpret the parameters - any state migration that is needed will still occur.

This is a great point, and one I have been trying to get to the bottom of too! I'd like to get your thoughts on two points here:

1. @timroes mentioned that this could be a problem for a plugin like dashboard whose state object can exceed 2MB when serialised. Another way to approach this could be to inject the state into the page using chromium AND support fetches _if_ the redirect app sees `jobId` and `locatorOffset` in the query params (see point 2).
2. A nice thing with fetching at execute time is we have very short URLs for generating reports that we could expose elsewhere in Kibana in addition to enabling users to click a link in the Reporting UI to go to where Chromium would go. Currently this is more of a nice-to-have though.

Perhaps we can add this functionality in a following PR and for the initial version we could just inject the state into the page.

Reply:
> mentioned that this could be a problem for a plugin like dashboard whose state object can exceed 2MB when serialised.

This is a [real issue](https://github.com/elastic/kibana/issues/20037) and [has blocked users](https://github.com/elastic/kibana/issues/27728#issuecomment-516159837) from doing things they want to do, when using Watcher to automate a PDF report. I worked on https://github.com/elastic/kibana/pull/45603 to unblock users from having too long URLs when they generate a report interactively, however POST URLs needed for automation can still be too long. I think the concern of @timroes stands to reason here, and gives us an opportunity to provide a better fix for this old issue.

> Another way to approach this could be to inject the state into the page using chromium

Chromium wouldn't be able to help here. The state can be injected into the page, but the app would have to be responsible for parsing it. 

The code that reporting can inject into the page doesn't have access to any Kibana APIs, just the data that's in the DOM. It would be able to read it, but it couldn't relay it to any app.

> A nice thing with fetching at execute time is we have very short URLs for generating reports that we could expose elsewhere in Kibana in addition to enabling users to click a link in the Reporting UI to go to where Chromium would go. Currently this is more of a nice-to-have though.

This is interesting. Would there have to be a persisted saved object for this to work?